docs/manual/codespaces.md

```diff
@@ -0,0 +1,273 @@
+- don't need to run the labs 24/7
+- want to demo labs on-the-go without managing the local environment
+
+You can select which GitHub machine type you want to use for your project; each machine type is characterized by the amount of CPUs/RAM/Storage that it is equipped with, and based on that you can calculate how many cpu-hours you would consume running a lab with a chosen machine type.
+
+![pic](https://gitlab.com/rdodin/pics/-/wikis/uploads/fc7bfa89c921f965ce61f6eb5e86f312/image.png){.img-shadow}
+
+<small>If you need more than 120 cpu-hours, you can pay for the additional hours, but the cost is reasonable, and you can always stop the environment when you don't need it to save the quota.</small>
+
+Your cpu-hours counter is reset at the beginning of each month, so you can use the free plan every month. And by default you have a $0 spending limit, so you won't be charged unless you explicitly allowed it. Good!
+
+#### Control panel
+
+Whenever you need to check what Codespaces environments you have running or created, you can do it in the [Codespaces panel][codespace-panel].
+
+![panel](https://gitlab.com/rdodin/pics/-/wikis/uploads/73aa0c2154959d4b25f2b1c2dbd1380b/image.png){.img-shadow}
+
+The panel allows you to see and interact with the available Codespaces environments, including starting, stopping, and deleting them. You can also check what repositories are associated with each environment which is useful for a Containerlab user to quickly identify the lab environments.
+
+/// details | Codespaces settings
+
+Codespaces expose a bunch of per-user settings at the [github.com/settings/codespaces](https://github.com/settings/codespaces) page. The following settings are worth mentioning:
+
+//// define
+
+Secrets
+
+- Secrets allow you to store sensitive information that will be available to your Codespaces environments. You can use them to store API keys, passwords, and other sensitive data that you don't want to expose in your code.
+
+Setting Sync
+
+- To make codespaces env feel like home, you can sync your settings across all your Codespaces environments. This includes themes, keybindings, and other settings that you have configured in your local VS Code instance.
+
+Editor preference
+
+- You can choose if you want to run the codespaces in a browser, in a local VS Code instance, or via a bridge to a JetBrains IDE.
+
+////
+
+///
+
+#### Billing
+
+It is always a good idea to periodically check how much of the cpu-hours you've consumed and check the remaining quota. Your billing information is available in the [Billing settings][billing].
+
+![billing](https://gitlab.com/rdodin/pics/-/wikis/uploads/4bf6eb89bd3b4f2d4b05ecdc3ab22675/image.png){.img-shadow}
+
+The screenshot shows that 10 cpu-hours out of 120 available were consumed in the current month' period and the codespaces environments occupy 8.15 GB of storage out of 15 GB included. So far it is all well within the free tier limits.[^5]
```

Comment:
The space thingy is complex, [the way gh counts that is mystical](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#about-billing-for-storage-usage), and I am not yet sure how to read the data usage properly.

If I launch the codespaces from this repo https://github.com/srl-labs/srl-telemetry-lab I get a codespaces that measures as 7.39GB

<img width="1168" alt="image" src="https://github.com/srl-labs/containerlab/assets/5679861/f2c6ac15-fa6f-4d21-b925-56b716d60452">

If I enter into this VM and use `df -h` than I see that 8GB.

If I start the lab from here https://github.com/srl-labs/srlinux-vlan-handling-lab/ which uses the same image for codespaces, it reports 3.3GB. And I don't quite get it at this moment...

<img width="1162" alt="image" src="https://github.com/srl-labs/containerlab/assets/5679861/a46f272d-9bbf-4fef-b01c-4244a031f845">

Reply:
I think the best way to find out what's happening is to go to your profile -> Billings and Plans and click "Get usage report" right to "Usage this month" section. Github will send and email with a link to a CSV explaining all the billing calculations. Also expand Storage section under Codespaces and make sure you don't see "Prebuild Storage" used.  
Here is my usage as of today (feel free to use it as a healthy example):
<img width="1145" alt="image" src="https://github.com/srl-labs/containerlab/assets/15916030/f60dbd4d-5952-4f08-b606-52ef26243c0b">
As you can see, CPU is ahead of storage and that's what I expect at all times for a proper Codespace configuration.  
And here is an example Codespace:
<img width="1310" alt="image" src="https://github.com/srl-labs/containerlab/assets/15916030/3b5adb4a-92eb-4c97-a93f-0f5114db8fff">
My understanding is that if I run this Codespace constantly for a month - I'd be charged for 3.36 GB-month. However I always stop it quick and storage usage is at minimum. Prebuild (if enabled) would change that and start consuming storage 24/7. But we don't need it as we are relying on Github packages and pre-build container images that are free for public repos.  

Here is the relevant quote from GH docs:

For example, if you have one codespace that uses 100 GB of storage and has existed for one hour you will have used 100 / (24 * 30) = 0.1388 GB-months of storage in a 30-day month. If your use of GitHub Codespaces during a 30-day month consists of two 100 GB codespaces that both existed for three full days then there will be 24 * 3 hourly reports for the storage of these codespaces, giving a total of (24 * 3) * 200 / (24 * 30) = 20 GB-months.

So, I strongly suspect you have prebuilds enabled. Although it could be something else that will be visible in CSV report anyway.

Reply:
The report shows some numbers, and I guess they totalling to the 8GB of storage quota consumed, but as I mentioned, the calculation is complex to me

<img width="817" alt="image" src="https://github.com/srl-labs/containerlab/assets/5679861/6a8ba746-8722-4ee8-b8be-f44cad184042">

Reply:
OK. Looks like you have not deleted some Codespaces after starting them, that would explain storage utilization on some days without compute.  
TLDR; I'd add a section to the doc about deleting Codespaces from https://github.com/codespaces after the workshop and maybe suggest lowering the retention period to 1 day.
I assume 901 hours on 11.06 is a bug. That would use all free compute if true.

Reply:
yep, definitely some bugs in calculation for a couple of runs.

I added a tip about 1 day retention period - [e1f2b7f](https://github.com/srl-labs/containerlab/pull/2089/commits/e1f2b7fcbcc770b4d38e8c1cdef155bef231f08b)